New example - filling up a glass of water (#279)
serverlessworkflow/specification: examples/README.md

</td>
</tr>
</table>

### Filling a glass of water

#### Description

In this example we showcase the power of [expression functions](../specification.md#Using-Functions-For-Expression-Evaluation).
Our workflow definition is assumed to have the following data input:

```json
{
  "counts": {
    "current": 0,
    "max": 10
  }
}
```
 

> Our workflow simulates filling up a glass of water one "count" at a time until "max" count is reached which
> represents our glass being full.
> Each time we increment the current count, the workflow checks if we need to keep refilling the glass.
> If the current count reaches the max count, the workflow execution ends.
> To increment the current count, the workflow invokes the "IncrementCurrent" expression function.
> It's results are then merged back into the state data according to the "toStateData" property of the event data filter.

Reviewer: ```suggestion
Our workflow simulates filling up a glass of water one "count" at a time until "max" count is reached which
represents our glass is full.
Each time we increment the current count, the workflow checks if we need to keep refilling the glass.
If the current count reaches the max count, the workflow execution ends.
To increment the current count, the workflow invokes the "IncrementCurrent" expression function.
Its results are then merged back into the state data according to the "toStateData" property of the event data filter.
```
